tests/performance/density-kwok.go

```diff
@@ -0,0 +1,227 @@
+package performance
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+const (
+	nodeCount           = 100
+	vmCount             = 1000
+	vmBatchStartupLimit = 5 * time.Minute
+)
+
+var _ = FSIGDescribe("[KWOK] Control Plane Performance Density Testing using kwok", func() {
+	var (
+		kubevirtClient kubecli.KubevirtClient
+		k8sClient      *kubernetes.Clientset
+		startTime      time.Time
+	)
+
+	artifactsDir, _ := os.LookupEnv("ARTIFACTS")
+
+	BeforeEach(func() {
```

Comment:
question: Is the workflow here:

1. create fake nodes
2. create fake VMIs
3. delete fake nodes
4. create fake nodes
5. create fake VMs
6. delete fake nodes
?

Reply:
Yes, that's correct

Reply:
@alaypatel07 Updated test to have this flow